pennylane/numpy/tensor.py

```diff
@@ -197,6 +197,23 @@ def __hash__(self):
 
         raise TypeError("unhashable type: 'numpy.tensor'")
 
+    def __reduce__(self):
+        # Called when pickling the object.
+        # Numpy ndarray uses __reduce__ instead of __getstate__ to prepare an object for
+        # pickling. self.requires_grad needs to be included in the tuple returned by
+        # __reduce__ in order to be preserved in the unpickled object.
+        reduced_obj = super().__reduce__()
+        # The last (2nd) element of this tuple holds the data. Add requires_grad to this:
+        full_reduced_data = reduced_obj[2] + (self.requires_grad,)
+        return (reduced_obj[0], reduced_obj[1], full_reduced_data)
+
+    def __setstate__(self, reduced_obj) -> None:
+        # Called when unpickling the object.
+        # Set self.requires_grad with the last element in the tuple returned by __reduce__:
+        self.requires_grad = reduced_obj[-1]

```

Comment:
```suggestion
        # pylint: disable=attribute-defined-outside-init
        self.requires_grad = reduced_obj[-1]
```

Safe to disable this lint warning :)

Reply:
Oh good, I was wondering about that, since there's no `__init__`. Thanks.